File: cpp/src/p5/main.cc
#include <iostream>

bool divisible_by_all(int n) {
  for (int i = 2; i <= 20; ++i) {
    if (n % i != 0) {
      return false;
    }
  }
  return true;
}

int main() {
  int n = 20;
  for(;;) {
    if (divisible_by_all(n)) {
      break;
    }
    n++;
  }

  std::cout << n << "\n";
}

File: golang/src/p5/main.go
// 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
// What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?

package main

import "fmt"

func main() {
	n := 20
	for {
		if divisible_by_all(n) {
			break
		}
		n++
	}

	fmt.Println(n)
}

func divisible_by_all(n int) bool {
	for i := 2; i <= 20; i++ {
		if n%i != 0 {
			return false
		}
	}

	return true
}
